Python/opcode_metadata.h

```diff
@@ -0,0 +1,184 @@
+    [GET_YIELD_FROM_ITER] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [FOR_ITER] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [FOR_ITER_LIST] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [FOR_ITER_TUPLE] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [FOR_ITER_RANGE] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [FOR_ITER_GEN] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [BEFORE_ASYNC_WITH] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [BEFORE_WITH] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [WITH_EXCEPT_START] = { 4, 5, DIR_NONE, DIR_NONE, DIR_NONE },
+    [PUSH_EXC_INFO] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [LOAD_ATTR_METHOD_WITH_VALUES] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [LOAD_ATTR_METHOD_WITH_DICT] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [LOAD_ATTR_METHOD_NO_DICT] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [LOAD_ATTR_METHOD_LAZY_DICT] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [CALL_BOUND_METHOD_EXACT_ARGS] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [KW_NAMES] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [CALL] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [CALL_PY_EXACT_ARGS] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [CALL_PY_WITH_DEFAULTS] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [CALL_NO_KW_TYPE_1] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [CALL_NO_KW_STR_1] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [CALL_NO_KW_TUPLE_1] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [CALL_BUILTIN_CLASS] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [CALL_NO_KW_BUILTIN_O] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [CALL_NO_KW_BUILTIN_FAST] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [CALL_BUILTIN_FAST_WITH_KEYWORDS] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [CALL_NO_KW_LEN] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [CALL_NO_KW_ISINSTANCE] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [CALL_NO_KW_LIST_APPEND] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [CALL_NO_KW_METHOD_DESCRIPTOR_O] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [CALL_METHOD_DESCRIPTOR_FAST_WITH_KEYWORDS] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [CALL_NO_KW_METHOD_DESCRIPTOR_NOARGS] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [CALL_NO_KW_METHOD_DESCRIPTOR_FAST] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [CALL_FUNCTION_EX] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [MAKE_FUNCTION] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [RETURN_GENERATOR] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [BUILD_SLICE] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [FORMAT_VALUE] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [COPY] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [BINARY_OP] = { 2, 1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [SWAP] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [EXTENDED_ARG] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
+    [CACHE] = { -1, -1, DIR_NONE, DIR_NONE, DIR_NONE },
```

Comment:
END_FOR is missing, I think we need something to handle macros. 

Also, not that missing entries default to 0. Would be safer if they defaulted to -1.

Reply:
> Also, note that missing entries default to 0. Would be safer if they defaulted to -1.

The code generator doesn't know the numeric values of the opcodes (yet) so it doesn't have enough info to generate explicit entries filled with `-1` for missing values. The best we can do is add an additional flag `valid_entry`.